src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -85,10 +95,16 @@ internal override async Task<SyntaxNode> GetTypeDeclarationAsync(Document docume
                 }
             }
 
-            var updatedTypeDeclaration = typeDeclaration.WithBaseList(updatedBaseList.WithAdditionalAnnotations(Formatter.Annotation));
-            var updatedRoot = root.ReplaceNode(root.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single(), updatedTypeDeclaration);
-            var solutionWithOriginalTypeUpdated = solutionWithFormattedInterfaceDocument.WithDocumentSyntaxRoot(docId, updatedRoot, PreservationMode.PreserveIdentity);
-            return solutionWithOriginalTypeUpdated;
+            return typeDeclaration.WithBaseList(updatedBaseList.WithAdditionalAnnotations(Formatter.Annotation));
+        }
+
+        private TypeSyntax GetTypeSyntaxFromNamedSymbol(INamedTypeSymbol extractedInterfaceSymbol)
+        {
+            return extractedInterfaceSymbol.TypeParameters.Any()
+                ? SyntaxFactory.GenericName(
+                    SyntaxFactory.Identifier(extractedInterfaceSymbol.Name),
+                    SyntaxFactory.TypeArgumentList(SyntaxFactory.SeparatedList(extractedInterfaceSymbol.TypeParameters.Select(p => SyntaxFactory.ParseTypeName(p.Name)))))
+                : SyntaxFactory.ParseTypeName(extractedInterfaceSymbol.Name);
```

Comment:
here as well. Thse are just identifiers right?  So this should just be `SyntaxFactory.IdentifierName(...)`